File: three_sphere.c
#include <stdio.h>
#include <stdlib.h>
#include <math.h>
#include <time.h>

int main(void){
  int niter=100000;
  srand((unsigned)time(NULL)); 
  
  double pi=asin(1e0)*2e0;
  double sum_z=0e0;
  int n_in=0;
  /*************/
  /* Main loop */
  /*************/
  for(int iter=1;iter<niter+1;iter++){

    double x = (double)rand()/RAND_MAX;
    double y = (double)rand()/RAND_MAX;
    if(x*x+y*y < 1e0){
      n_in=n_in+1;
      double z=sqrt(1e0-x*x-y*y);
      sum_z=sum_z+z;
    }
    printf("%d   %.10f\n",iter,sum_z/n_in*2e0*pi);}
}

File: three_sphere.py
import numpy as np
# seed of random numbers is the system time by default

NITER = 100000

PI = np.pi #numpy knows PI.
sum_z=0.0
n_in = 0
#################
### Main Loop ###
#################
for iter in range(NITER):
    x = np.random.uniform(0, 1)
    y = np.random.uniform(0, 1)
    if(x*x+y*y < 1.0):
        #n_in=n_in+1
        n_in += 1
        z = np.sqrt(1.0-x*x-y*y)
        #sum_z = sum_z + z
        sum_z += z
    if(n_in > 0):#We get error if N_in=0
        print(iter+1,sum_z/n_in*2e0*PI)


